src/components/Search/Search.js

```diff
@@ -1,147 +1,103 @@
-import React, { useEffect, useMemo, useRef, useState } from 'react';
-import { useSelector } from 'react-redux';
-import Autosuggest from 'react-autosuggest';
-import {
-  FaSearch,
-  FaTimes,
-  FaChevronCircleDown,
-  FaChevronCircleUp,
-} from 'react-icons/fa';
+import OLVectorLayer from 'ol/layer/Vector';
+import OLVectorSource from 'ol/source/Vector';
+import PropTypes from 'prop-types';
+import React, { useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { FaChevronCircleDown, FaChevronCircleUp } from 'react-icons/fa';
 import { useTranslation } from 'react-i18next';
+import RTSearch from 'react-transit/components/Search';
 
 import SearchToggle from './SearchToggle';
 
-import './Search.scss';
+import 'react-transit/components/Search/Search.scss';
 
-function Search() {
-  const [suggestions, setSuggestions] = useState([]);
-  const [value, setValue] = useState('');
-  const map = useSelector(state => state.app.map);
-  const searchService = useSelector(state => state.app.searchService);
-  const { t } = useTranslation();
+const propTypes = {
+  apiKey: PropTypes.string.isRequired,
+};
 
-  const componentIsMounted = useRef(true);
-  useEffect(() => {
-    return () => {
-      componentIsMounted.current = false;
-    };
-  }, []);
+function Search({ apiKey }) {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const { layerService, map, searchEngines } = useSelector(state => ({
+    layerService: state.app.layerService,
+    map: state.app.map,
+    searchEngines: state.app.activeTopic.searchEngines,
+  }));
```

Comment:
this generates always a rerender of the component . I will separate in 3 useSelector instead.